### PR TITLE
1173 diagram annotations

### PR DESCRIPTION
**Pull request by philip-alldredge**
Implements #1173.

### CODE REVIEW

**Comment by RyanMcilnay** [comment]
@philip-alldredge 

Copy Paste Bug with Note:
1. Select Note
2. Copy
3. Select a feature, flow spec, or connection (possibly more than this)
4. Paste

Stacktrace:
[note_copy_paste_exception_stacktrace.txt](https://github.com/osate/osate2/files/2446530/note_copy_paste_exception_stacktrace.txt)

**Comment by philip-alldredge** [comment]
@RyanMcilnay the notes issues should be fixed. I've also updated the branch to the latest develop.

File: ge/org.osate.ge/src/org/osate/ge/internal/util/EObjectDocumentationUtil.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		// Get the resource service provider
+		final IResourceServiceProvider resourceServiceProvider = resource.getResourceServiceProvider();
+		if (resourceServiceProvider == null) {
+			return null;
+		}
+
+		// Get the documentation provider
+		final IEObjectDocumentationProvider docProvider = resourceServiceProvider
+				.get(IEObjectDocumentationProvider.class);
+		return docProvider;
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/util/EObjectDocumentationUtil.java]: Possibly change to return resourceServiceProvider.get(IEObjectDocumentationProvider.class);

Just cleanup, depends on preference.

**Comment by RyanMcilnay** [comment]
Selecting multiple notes with "Note" properties section open.

1. Select a note
2. Open "Note" property section
3. Hold control and select another note

[multiple_note_selection.txt](https://github.com/osate/osate2/files/2447045/multiple_note_selection.txt)

**Comment by philip-alldredge** [comment]
@RyanMcilnay the described issues should be fixed.

File: ge/org.osate.ge/src/org/osate/ge/internal/ui/properties/NotePropertySection.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+
+		noteField = getWidgetFactory().createText(composite, "", SWT.MULTI | SWT.WRAP | SWT.V_SCROLL);
+		noteField.setLayoutData(GridDataFactory.swtDefaults().align(SWT.FILL, SWT.FILL).grab(true, true)
+				.hint(100, 100).create());
+
+		// Create the button that will update the diagram. This button won't actually do anything because the diagram is updated when the focus is lost.
+		// However, it is useful to have it so that the user can easily see the result of the change without without having to change property tabs/selection.
+		saveBtn = getWidgetFactory().createButton(composite, "Save", SWT.PUSH);
+
+		noteField.addFocusListener(new FocusListener() {
```
Comment (RyanMcilnay) [inline ge/org.osate.ge/src/org/osate/ge/internal/ui/properties/NotePropertySection.java]: Change to FocusAdapter

Comment (philip-alldredge) [reply ge/org.osate.ge/src/org/osate/ge/internal/ui/properties/NotePropertySection.java]: Fixed

File: ge/org.osate.ge.doc/docs/user_guide.md [hunk 3]
```diff
@@ -247,6 +247,15 @@ If a a model element is renamed from the AADL text editor without using the *Ren
 2. Update the desired model references in the *New Reference* column.
 3. Select OK. The diagram elements will be restored.
 
+## Annotations and Documentation
+The following sections describe the graphical editors support for annotations and AADL documentation.
+
+### Annotations
+Diagrams can be annotated using the tools available in the *Annotation* section of the palette. The *Note* option allows add text notes to the diagram. Notes are part of the diagram and are not part of the AADL model. The *Note Reference* option can be used to associate a note with another element in the diagram.
```
Comment (RyanMcilnay) [inline ge/org.osate.ge.doc/docs/user_guide.md]: "The *Note* option allows add text notes..." should change to "... allows for adding text notes..." or something similar.

**Comment by philip-alldredge** [comment]
@RyanMcilnay the user guide issue should be fixed.

**Comment by RyanMcilnay** [comment]
@philip-alldredge 
Based off discussion:
1. *Documentation* property view has unwanted new lines on block comments.
2. Possibly remove caret in *Documentation* text field.

**Comment by philip-alldredge** [comment]
@RyanMcilnay  issues have been fixed. Documentation field has been disabled to indicate that it is read-only.

**Comment by RyanMcilnay** [comment]
@philip-alldredge 
Because Documentation field is disabled, the scroll bar no longer is selectable.

**Comment by philip-alldredge** [comment]
@RyanMcilnay I have reverted the disabling of the field. If it causes confusion for users, we can revisit.

**Pull request merged**